sql/core/src/main/scala/org/apache/spark/sql/util/ContinuousQueryListener.scala

```diff
@@ -65,11 +65,33 @@ object ContinuousQueryListener {
   trait Event
 
   /** Event representing the start of a query */
-  class QueryStarted private[sql](val query: ContinuousQuery) extends Event
+  class QueryStarted private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
   /** Event representing any progress updates in a query */
-  class QueryProgress private[sql](val query: ContinuousQuery) extends Event
+  class QueryProgress private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
-  /** Event representing that termination of a query */
-  class QueryTerminated private[sql](val query: ContinuousQuery) extends Event
+  /**
+   * Event representing that termination of a query
+   *
+   * @param queryInfo
+   * @param exception The excpetion information of the [[ContinuousQuery]] if any. Otherwise, it
+   *                  will be `None`.
+   */
+  class QueryTerminated private[sql](
+      val queryInfo: ContinuousQueryInfo,
+      val exception: Option[String]) extends Event
```

Comment:
I think we want the stacktrace too as well as how far it made it before it failed.  It might be okay to serialize the exception itself, we just need to make sure its not pulling in the whole query.